core/domain/skill_services.py

```diff
@@ -230,6 +232,239 @@ def get_all_skill_summaries():
+                assigned_skill_ids[skill_summary_id]['classroom_name'])
+
+    return augmented_skill_summaries, next_cursor, more, total_count
+
+
+def filter_skills_by_status(augmented_skill_summaries, status):
+    """Returns the skill summary dicts after filtering by status.
+
+    Args:
+        augmented_skill_summaries: list(AugmentedSkillSummary). The list
+        of augmented skill summaries.
+        status: str. The status of the skill.
+
+    Returns:
+        list(AugmentedSkillSummary). The list of AugmentedSkillSummaries
+        matching the given status.
+    """
+
+    if not status or status == constants.SKILL_STATUS_OPTIONS['ALL']:
+        return augmented_skill_summaries
+
+    elif status == constants.SKILL_STATUS_OPTIONS['UNASSIGNED']:
+        unassigned_augmented_skill_summaries = []
+        for augmented_skill_summary in augmented_skill_summaries:
+            if not augmented_skill_summary.topic_name:
+                unassigned_augmented_skill_summaries.append(
+                    augmented_skill_summary)
+
+        return unassigned_augmented_skill_summaries
+
+    elif status == constants.SKILL_STATUS_OPTIONS['ASSIGNED']:
+        assigned_augmented_skill_summaries = []
+        for augmented_skill_summary in augmented_skill_summaries:
+            if augmented_skill_summary.topic_name:
+                assigned_augmented_skill_summaries.append(
+                    augmented_skill_summary)
+
+        return assigned_augmented_skill_summaries
+
+
+def filter_skills_by_classroom(augmented_skill_summaries, classroom_name):
+    """Returns the skill summary dicts after filtering by classroom_name.
+
+    Args:
+        augmented_skill_summaries: list(AugmentedSkillSummary).
+            The list of augmented skill summaries.
+        classroom_name: string. The name of the classroom.
+
+    Returns:
+        list(AugmentedSkillSummary). The list of augmented
+            skill summaries with the given classroom name.
+    """
+
+    if not classroom_name or classroom_name == 'All':
+        return augmented_skill_summaries
+
+    augmented_skill_summaries_with_classroom_name = []
+    for augmented_skill_summary in augmented_skill_summaries:
+        if (augmented_skill_summary.classroom_name and
+                augmented_skill_summary.classroom_name == classroom_name):
+            augmented_skill_summaries_with_classroom_name.append(
+                augmented_skill_summary)
+
+    return augmented_skill_summaries_with_classroom_name
+
+
+def is_keyword_present_in_skill(augmented_skill_summary, keywords):
+    """Returns whether the keywords match the skill description.
+
+    Args:
+        augmented_skill_summary: list(AugmentedSkillSummary).
+            The augmented skill summaries.
+        keywords: list(str). The keywords to match.
+
+    Returns:
+        bool. A boolean indicating if any of the keywords exist in the given
+        augmented skill summary.
+    """
+    for keyword in keywords:
```

Comment:
You can do `return any(...)`

Also I think you can probably inline this? (once it becomes a one-line command)

Reply:
Done.